packages/evm/src/opcodes/functions.ts

```diff
@@ -593,19 +594,35 @@ export const handlers: Map<number, OpHandler> = new Map([
   // 0x40: BLOCKHASH
   [
     0x40,
-    async function (runState) {
+    async function (runState, common) {
       const number = runState.stack.pop()
 
-      const diff = runState.interpreter.getBlockNumber() - number
-      // block lookups must be within the past 256 blocks
-      if (diff > BIGINT_256 || diff <= BIGINT_0) {
-        runState.stack.push(BIGINT_0)
-        return
-      }
+      if (common.isActivatedEIP(2935)) {
+        if (number >= runState.interpreter.getBlockNumber()) {
+          runState.stack.push(BIGINT_0)
+          return
+        }
 
-      const block = await runState.blockchain.getBlock(Number(number))
+        const historyAddress = Address.fromString(
```

Comment:
Very optional and not directly PR related, just since I noted that many conversion steps:
Would it make sense to add a `Address.fromBigInt()` additional constructor to `Address` where we can skip one (the to hex string one) conversion step? 🤔 

(decide for yourself if a) yes or no and b) if to add here)

Reply:
Note: I had to convert bigInt -> Address because in our Common the only allowed `v` values for the `param`s of EIPs (such as `historyStorageAddress`) is bigint. I am not sure if we want to add the `fromBigInt` method because it seems a bit error prone to me: if you delete some numbers, you get a number which in hex string would be less than 20 bytes / 40 characters, but it is still a correct `bigint` - it sounds slightly unsafe to me. (I think we can discuss this but we should address this in a different PR if we want this)

Reply:
👍